packages/babel-plugin-remove-graphql-queries/src/index.js

```diff
@@ -36,13 +39,61 @@ function getGraphQLTag(path) {
 export default function({ types: t }) {
   return {
     visitor: {
-      TaggedTemplateExpression(path, state) {
-        const ast = getGraphQLTag(path)
+      Program(path, state) {
+        const nestedJSXVistor = {
+          JSXIdentifier(path2) {
+            if (
+              [`production`, `test`].includes(process.env.NODE_ENV) &&
+              path2.isJSXIdentifier({ name: `StaticQuery` })
+            ) {
+              const identifier = t.identifier("staticQueryData")
+              const filename = state.file.opts.filename
+              const shortResultPath = `public/static/d/${this.queryHash}.json`
+              const resultPath = nodePath.join(process.cwd(), shortResultPath)
+              // Add query
+              path2.parent.attributes.push(
+                t.jSXAttribute(
+                  t.jSXIdentifier(`data`),
+                  t.jSXExpressionContainer(identifier)
+                )
+              )
+              // Add import
+              const importDefaultSpecifier = t.importDefaultSpecifier(
+                identifier
+              )
+              const importDeclaration = t.importDeclaration(
+                [importDefaultSpecifier],
+                t.stringLiteral(
+                  filename
+                    ? nodePath.relative(
+                        nodePath.parse(filename).dir,
+                        resultPath
+                      )
+                    : shortResultPath
+                )
+              )
+              path.unshiftContainer("body", importDeclaration)
+            }
+          },
+        }
 
-        if (!ast) return null
+        path.traverse({
+          TaggedTemplateExpression(path2, state) {
+            const { ast, text, hash } = getGraphQLTag(path2)
 
-        path.replaceWith(t.StringLiteral(`** extracted graphql fragment **`))
-        return null
+            if (!ast) return null
+
+            const queryHash = hash.toString()
+            const query = text
+
+            // Replace the query with the hash of the query.
+            path2.replaceWith(t.StringLiteral(queryHash))
```

Comment:
why not replace with a `require` to the data itself, you could avoid the context bit in that case maybe?

Reply:
Because we moved to loading query results over websockets in https://github.com/gatsbyjs/gatsby/pull/4555 as it's way way faster than relying on webpack to hot reload data.